tests2/gui/CMakeLists.txt

```diff
@@ -13,26 +13,16 @@
 #  You should have received a copy of the GNU General Public License
 #  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-include_directories(
-        ${CMAKE_CURRENT_SOURCE_DIR}
-        ${CMAKE_CURRENT_BINARY_DIR})
-
 set(TEST_LIBRARIES keepassx_core)
-set(TEST_GUI_SUPPORT_SOURCES
-        main.cpp
+set(TEST_GUI_SOURCES
         ../../tests/util/TemporaryFile.cpp
+        main.cpp
         FixtureBase.cpp
-        FixtureWithDb.cpp)
-add_library(TEST_GUI_SUPPORT STATIC ${TEST_GUI_SUPPORT_SOURCES})
-target_link_libraries(TEST_GUI_SUPPORT Qt5::Core Qt5::Widgets Qt5::Test Catch2::Catch2)
-
-# Add new unit tests here.
-# Naming convention: word-cased separated by dash.
-# Avoid naming with "test" as obvious and useless prefix.
-add_unit_test(NAME pass-phrase-generator-gui
-        SOURCES TestPassphraseGeneratorGui.cpp
-        LIBS TEST_GUI_SUPPORT ${TEST_LIBRARIES})
+        FixtureWithDb.cpp
+        TestPassphraseGeneratorGui.cpp
+        TestTotpGui.cpp)
+add_executable(TestGui ${TEST_GUI_SOURCES})
+target_link_libraries(TestGui ${TEST_LIBRARIES} Qt5::Test Catch2::Catch2)
 
-add_unit_test(NAME totp-gui
-        SOURCES TestTotpGui.cpp
-        LIBS TEST_GUI_SUPPORT ${TEST_LIBRARIES})
+add_test(TEST_GUI TestGui)
```

Comment:
With this solution the test run output looks like (I temporary disabled the legacy tests and run only Catch2 ones):
```
Running tests...
Test project /home/dmytro/Desktop/github/fork/keepassxc/build
    Start 1: passphrase-generator
1/2 Test #1: passphrase-generator .............   Passed    0.06 sec
    Start 2: TEST_GUI
2/2 Test #2: TEST_GUI .........................   Passed   14.50 sec

100% tests passed, 0 tests failed out of 2
```

Meanwhile the original output was like:
```
Running tests...
Test project /home/dmytro/Desktop/github/fork/keepassxc/build
    Start 1: passphrase-generator
1/3 Test #1: passphrase-generator .............   Passed    0.06 sec
    Start 2: pass-phrase-generator-gui
2/3 Test #2: pass-phrase-generator-gui ........   Passed    4.44 sec
    Start 3: totp-gui
3/3 Test #3: totp-gui .........................   Passed   10.58 sec

100% tests passed, 0 tests failed out of 3
```
As for me the aggregated `TEST_GUI` is not informative and does not provide any clue how many GUI tests run, how much time each test takes, which tests were failed, and finally which GUI tests we have at all.

I would keep the `add_unit_test` usage, the same way as we already have for regular unit tests. I don't see any benefits to use aggregated `TEST_GUI`.

Reply:
There is a better middle ground, I simply merged all tests into one executable. You are looking at raw ctest output. When I run the tests in visual studio I see each and every catch2 test and the status. You might need to add a flag or invoke the tests differently.

Reply:
Yes, I know an IDE runs tests separately but I am talking about raw console logs which are only available in CI pipeline. Wouldn't be convenient for troubleshooting to have clear log with separate tests?

Reply:
If there is an error or failure, ctest should output where that occurs. Otherwise, the output is irrelevant and ignored. Checking for slow tests and other things like that are fine locally in an ide. Favor overall compile speed and test execution over verbose successful test run output.

Reply:
Got it, ok, let's keep it.